compiler/src/iree/compiler/Dialect/Stream/Transforms/DumpStatistics.cpp

```diff
@@ -437,18 +437,32 @@ static void dumpExecutionCSVTable(const UsageInfo &usageInfo,
           os << "\n";
         })
         .Case<IREE::Stream::CmdDispatchOp>([&](auto op) {
-          auto workload = op.workgroup_count();
-          APInt workloadX;
-          APInt workloadY;
-          APInt workloadZ;
-          matchPattern(workload[0], m_ConstantInt(&workloadX));
-          matchPattern(workload[1], m_ConstantInt(&workloadY));
-          matchPattern(workload[2], m_ConstantInt(&workloadZ));
+          // NOTE: we only use up to the first 3 workload operands as we need
+          // to keep the CSV fixed-width. We'll still calculate the element
+          // count using all of them, however all these numbers are just guesses
+          // anyway.
+          APInt workloadX, workloadY, workloadZ;
+          auto workload = op.workload();
+          if (workload.size() > 0) {
+            matchPattern(workload[0], m_ConstantInt(&workloadX));
+            if (workload.size() > 1) {
+              matchPattern(workload[1], m_ConstantInt(&workloadY));
+              if (workload.size() > 2) {
+                matchPattern(workload[2], m_ConstantInt(&workloadZ));
```

Comment:
I guess we could also replace `"X","Y","Z"` in the schema with a String like `Workload` (e.g. a semicolon delimited list of numbers of arbitrary length)

Reply:
True; my original thought with the x,y,z was that it'd be easy to pull out and sort/group the dispatches by workgroup count, but it's not super hard to do that with semicolon-delimited values and it was never really workgroup count in the first place so the usage was questionable. Changed to `"1;2;?;4"` for something like `dispatch[%c1, %c2, %dynamic, %c4]`.